sp_peaks/slics.py

```diff
@@ -69,3 +69,116 @@ def read_catalogue(file_path, all_col=True):
     dat = ascii.read(file_path, names=col_names, include_names=include_names)
 
     return dat
+
+import numpy as np
+
+def read_SLICS_cats(file_paths):
+    """Reads the information in the filename.
+
+    Parameters
+    ----------
+    file_paths : list of str (.txt file)
+        List containing the paths to the files to be processed
+
+    Returns
+    -------
+    data : recarray
+        Numpy recarray containing the information extracted from the file names
+
+    """
+
+    # Make empty recarray to store the data
+    data = np.recarray(len(file_paths), dtype=[('id', int), ('seed', 'U1'), ('bin', int), ('LOS', int), ('tile', int)])
+
+    # Iterate over the file paths and process each file
+    for i, file_path in enumerate(file_paths):
+        # Extract the file name from the file path
+        file_name = file_path.split("/")[-1]
+        
+        # Split file name into parts
+        file_parts = file_name.split("_")
+        
+        id = int(file_parts[2])
+        seed = file_parts[3]
+        unknown_number = int(file_parts[4])
+        bin = int(file_parts[5][3:])  # Extract the number after "Bin"
+        LOS = int(file_parts[6][3:])  # Extract the number after "LOS"
+        tile = int(file_parts[7][1:-4])  # Extract the number after "R"
+
+        # Assign the extracted data to the corresponding fields in the recarray
+        data[i]['id'] = id
+        data[i]['seed'] = seed
+        data[i]['bin'] = bin
+        data[i]['LOS'] = LOS
+        data[i]['tile'] = tile
+        # print(data[i])
+
+    return data 
+
+def read_SLICS_cosmo_params(file_path):
```

Comment:
i'd modify this function to look more like "read_catalog"

Reply:
The file that is to be read by this function is a .dat file (CosmoTable.dat) and I think it cannot be read with `ascii.read()`, so I could only do it like this. There should however be a better way to do this, I just don't know it.